contenteditable selector is inadequate, not compatible with HAML-generated HTML

According to http://haml.info/docs/yardoc/file.REFERENCE.html#boolean_attributes the selector at https://github.com/NYTimes/backbone.stickit/blob/13a08b67a236186bb41363bd4ca49de4fadba39f/backbone.stickit.js#L431 is incorrect. Should be `'[contenteditable]'`

well kinda...
```
$ npm test

> backbone.stickit@0.8.0 test {{.}}
> phantomjs test/vendor/runner.js test/index.html?noglobals=true

Took 536ms to run 371 tests. 370 passed, 1 failed.
Unsafe JavaScript attempt to access frame with URL about:blank from frame
with URL file://test/vendor/runner.js. Domains, protocols and ports must
match.

Unsafe JavaScript attempt to access frame with URL about:blank from frame
with URL file://test/vendor/runner.js. Domains, protocols and ports must
match.

Unsafe JavaScript attempt to access frame with URL about:blank from frame
with URL file://test/vendor/runner.js. Domains, protocols and ports must
match.

npm ERR! Test failed.  See above for more details.
npm ERR! not ok code 0
```

On 5 January 2015 at 21:30, Adam Krebs <notifications@github.com> wrote:

> Is npm test working for you?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NYTimes/backbone.stickit/issues/268#issuecomment-68762108>
> .
>

--- a/backbone.stickit.js
+++ b/backbone.stickit.js
@@ -428,7 +428,7 @@
   // ----------------
 
   Stickit.addHandler([{
-    selector: '[contenteditable="true"]',
+    selector: '[contenteditable]',
     updateMethod: 'html',
     events: ['input', 'change']
   }, {

--- a/test/bindData.js
+++ b/test/bindData.js
@@ -44,25 +44,28 @@ $(document).ready(function() {
     equal(model.get('water'), 'dasina');
   });
 
-  test('contenteditable', function() {
+  _([1,2]).each(function(subtest) {
+    var sel = '#test17-' + subtest;
 
-    model.set({'water':'<span>fountain</span>'});
-    view.model = model;
-    view.templateId = 'jst17';
-    view.bindings = {
-      '#test17': {
+    test('contenteditable-' + subtest, function() {
+      model.set({'water':'<span>fountain</span>'});
+      view.model = model;
+      view.templateId = 'jst17';
+      view.bindings = {};
+      view.bindings[sel] = {
         observe: 'water'
-      }
-    };
-    $('#qunit-fixture').html(view.render().el);
+      };
+      $('#qunit-fixture').html(view.render().el);
 
-    equal(view.$('#test17').html(), '<span>fountain</span>');
+      equal(view.$(sel).html(), '<span>fountain</span>');
 
-    model.set('water', '<span>evian</span>');
-    equal(view.$('#test17').html(), '<span>evian</span>');
+      model.set('water', '<span>evian</span>');
+      equal(view.$(sel).html(), '<span>evian</span>');
 
-    view.$('#test17').html('<span>dasina</span>').trigger('change');
-    equal(model.get('water'), '<span>dasina</span>');
+      view.$(sel).html('<span>dasina</span>').trigger('change');
+      equal(model.get('water'), '<span>dasina</span>');
+    });
+
   });
 
   test('checkbox', function() {

--- a/test/index.html
+++ b/test/index.html
@@ -98,7 +98,10 @@
 
 	<script id="jst16" type="text/jst"><select id="test16" multiple="true"></select></script>
 
-	<script id="jst17" type="text/jst"><div id="test17" contenteditable="true"></div></script>
+	<script id="jst17" type="text/jst">
+    <div id="test17-1" contenteditable="true"></div>
+    <div id="test17-2" contenteditable></div>
+  </script>
 
 	<script id="jst18" type="text/jst">
 		<input type="checkbox" class="boxes" value="fountain">

